test_integration/lib/utils/encoders.dart

```diff
@@ -15,7 +15,7 @@ Map<String, dynamic>? encodeMessageExtras(MessageExtras? extras) =>
 
 Map<String, dynamic> encodeMessage(Message message) => {
       'id': message.id,
-      'timestamp': message.timestamp!.toIso8601String(),
+      'timestamp': message.timestamp?.toIso8601String(),
```

Comment:
Why are we expecting timestamp to be null? A message in history should always bear a timestamp, right?

Reply:
True, but this was causing the failing/ flaky test. I didnt want to investigate the cause of the null in the timestamp since it's not an issue on the SDK side, but I really should report it. I'll take a look at this again next week

Reply:
Let's have an issue for this before landing this PR?

Reply:
Yup, here it is: https://github.com/ably/ably-flutter/issues/151